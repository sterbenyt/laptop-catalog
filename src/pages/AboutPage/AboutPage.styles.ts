import styled from 'styled-components';

export const Container = styled.div`
  max-width: 800px;
  margin: 40px auto;
  padding: 0 20px;
`;

export const Title = styled.h1`
  font-size: 36px;
  margin-bottom: 24px;
  text-align: center;
  color: #2a9d8f;
`;

export const Text = styled.p`
  font-size: 18px;
  line-height: 1.6;
  margin-bottom: 16px;
  color: #333;
`;

export {};
