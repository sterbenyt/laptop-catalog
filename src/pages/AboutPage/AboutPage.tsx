import React from 'react';
import * as S from './AboutPage.styles';

const AboutPage: React.FC = () => {
    return (
        <S.Container>
            <S.Title>Про нас</S.Title>
            <S.Text>
                Ласкаво просимо до нашого каталогу ноутбуків! Ми пропонуємо широкий вибір сучасних ноутбуків від провідних виробників.
                Наше завдання — допомогти вам знайти найкращий пристрій, який відповідає вашим потребам.
            </S.Text>
            <S.Text>
                Ми цінуємо якість, комфорт та зручність покупок. Якщо у вас є питання, звертайтесь до нашої служби підтримки.
            </S.Text>
        </S.Container>
    );
};

export default AboutPage;
