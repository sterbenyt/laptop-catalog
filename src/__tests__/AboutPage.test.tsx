import React from 'react';
import { render, screen } from '@testing-library/react';
import AboutPage from '../pages/AboutPage/AboutPage';
import '@testing-library/jest-dom';

describe('AboutPage', () => {
    it('renders without crashing', () => {
        render(<AboutPage />);
        expect(screen.getByText('Про нас')).toBeInTheDocument();
    });

    it('renders both paragraphs of text', () => {
        render(<AboutPage />);

        expect(
            screen.getByText(/Ласкаво просимо до нашого каталогу ноутбуків!/i)
        ).toBeInTheDocument();

        expect(
            screen.getByText(/Ми цінуємо якість, комфорт та зручність покупок/i)
        ).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { container } = render(<AboutPage />);
        expect(container).toMatchSnapshot();
    });
});
