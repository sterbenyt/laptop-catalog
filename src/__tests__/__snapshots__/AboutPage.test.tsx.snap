// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPage matches snapshot 1`] = `
<div>
  <div
    class="sc-bRKDuR fyqweN"
  >
    <h1
      class="sc-hvigdm cRlwMG"
    >
      Про нас
    </h1>
    <p
      class="sc-fhHczv iKzgOf"
    >
      Ласкаво просимо до нашого каталогу ноутбуків! Ми пропонуємо широкий вибір сучасних ноутбуків від провідних виробників. Наше завдання — допомогти вам знайти найкращий пристрій, який відповідає вашим потребам.
    </p>
    <p
      class="sc-fhHczv iKzgOf"
    >
      Ми цінуємо якість, комфорт та зручність покупок. Якщо у вас є питання, звертайтесь до нашої служби підтримки.
    </p>
  </div>
</div>
`;
